### PR TITLE
Adding feature python setup.py clean

### PR DESCRIPTION
**Pull request by dendisuhubdy**
adding feature of python setup.py clean where it will delete the designated build files stated in .clean which is Build/ build/ and others

### CODE REVIEW

File: versioneer.py [hunk 1]
```diff
@@ -1647,6 +1647,25 @@ def make_release_tree(self, base_dir, files):
...
+                    for filename in glob.glob(wildcard):
+                        try:
+                            os.remove(filename)
+                        except OSError:
+                            shutil.rmtree(filename, ignore_errors=True)
+
+            # It's an old-style class in Python 2.7...
+            distutils.command.clean.clean.run(self)
+    cmds["clean"] = cmd_clean
+
```
Comment (abergeron) [inline versioneer.py]: Don't put that in versioneer, it's going to the lost the next time we update it.

File: .clean [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+*.pyo
+*.egg-info
+dist
+setuptools*egg
+setuptools.pth
+distribute*egg
+distribute*tar.gz
+*.so
+*.o
+*.log
```
Comment (abergeron) [inline .clean]: Could this list be built into the code.  I'm not sure I like the idea of an external file list.

Comment (dendisuhubdy) [reply .clean]: Yes I could but it'll be ugly, one other choice is to load .gitignore than .clean

File: versioneer.py (hunk 1 above)
Comment (dendisuhubdy) [reply versioneer.py]: The thing is that the setup call at the end of `setup.py` uses cmds list from versioneer.

Comment (abergeron) [reply versioneer.py]: You can get the cmdclass from versioneer, add the custom clean command and then pass it to setup().

**Pull request merged**